internal/dcache/rpc/server/handler.go

```diff
@@ -46,44 +56,743 @@ var _ service.ChunkService = &ChunkServiceHandler{}
 
 // ChunkServiceHandler struct implements the ChunkService interface
 type ChunkServiceHandler struct {
-	locks *common.LockMap
-	// more fields can be added here as needed
+	locks   *common.LockMap
+	fsIDMap map[string]*rvInfo // map to store the fsID to rvInfo mapping
+}
+
+type rvInfo struct {
+	rvID     string       // rv0, rv1, etc.
+	cacheDir string       // cache dir path for this RV
+	mvMap    sync.Map     // sync map of MV id against mvInfo
+	mvCount  atomic.Int64 // count of MVs for this RV, this should be updated whenever a MV is added or removed from the sync map
+}
+
+type mvInfo struct {
+	syncInfo                  // sync info for this MV
+	peerRVs      []string     // sorted list of peer RVs for this MV
+	chunkOpsCnt  atomic.Int64 // count of chunk operations (get, put or remove) for this MV in the given RV
+	blockOpsFlag atomic.Bool  // block chunk operations for this MV in the given RV. This is done when the .sync dir contents are being moved to the regular MV dir
```

Comment:
why do you say "in the given RV", that's implicit and confusing.
we will not have same mv on two different RVs on the same node. That's against resiliency rules.